src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,2181 @@
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <test/jtx.h>
+#include <test/jtx/utility.h>
+#include <xrpl/protocol/Batch.h>
+#include <xrpl/protocol/Feature.h>
+#include <xrpl/protocol/STParsedJSON.h>
+#include <xrpl/protocol/Sign.h>
+#include <xrpl/protocol/TxFlags.h>
+#include <xrpl/protocol/jss.h>
+
+namespace ripple {
+namespace test {
+
+class Batch_test : public beast::unit_test::suite
+{
+    struct TestBatchData
+    {
+        std::string result;
+        std::string txHash;
+    };
+
+    struct TestSignData
+    {
+        int index;
+        jtx::Account account;
+    };
+
+    Json::Value
+    getTxByIndex(Json::Value jrr, std::uint8_t index)
+    {
+        for (auto const& txn : jrr[jss::result][jss::ledger][jss::transactions])
+        {
+            if (txn[jss::metaData][sfTransactionIndex.jsonName] == index)
+                return txn;
+        }
+        return {};
+    }
+
+    void
+    validateBatch(
+        jtx::Env& env,
+        TxID const& batchId,
+        std::vector<TestBatchData> const& batchResults)
+    {
+        Json::Value params;
+        params[jss::ledger_index] = env.current()->seq() - 1;
+        params[jss::transactions] = true;
+        params[jss::expand] = true;
+        auto const jrr = env.rpc("json", "ledger", to_string(params));
+        // std::cout << "jrr: " << jrr << std::endl;
+
+        // Validate the number of transactions in the ledger
+        auto const transactions =
+            jrr[jss::result][jss::ledger][jss::transactions];
+        BEAST_EXPECT(transactions.size() == batchResults.size() + 1);
+
+        // Validate ttBatch is correct index
+        auto const txn = getTxByIndex(jrr, 0);
+        BEAST_EXPECT(txn.isMember(jss::metaData));
+        Json::Value const meta = txn[jss::metaData];
+        BEAST_EXPECT(txn[sfTransactionType.jsonName] == "Batch");
+        BEAST_EXPECT(meta[sfTransactionResult.jsonName] == "tesSUCCESS");
+
+        // Validate the inner transactions
+        for (TestBatchData const& batchResult : batchResults)
+        {
+            Json::Value jsonTx;
+            jsonTx[jss::binary] = false;
+            jsonTx[jss::transaction] = batchResult.txHash;
+            jsonTx[jss::id] = 1;
+            Json::Value const jrr =
+                env.rpc("json", "tx", to_string(jsonTx))[jss::result];
+            BEAST_EXPECT(
+                jrr[jss::meta][sfTransactionResult.jsonName] ==
+                batchResult.result);
+            // BEAST_EXPECT(jrr[jss::meta][sfParentBatchID.jsonName] ==
+            // batchId);
+        }
+    }
+
+    Json::Value
+    addBatchTx(
```

Comment:
Why isn't this function (and the below ones) part of the `jtx` helper file? Seems like there's some overlap there.